### PR TITLE
Feat/show document data

### PR DESCRIPTION
**Pull request by mitrejcevski**
- Implemented document data preview 
 - Centralized user auth prompt into a reusable component 
 - Improved code strucure and formatting

### CODE REVIEW

File: identity/src/main/java/com/android/identity/PresentationHelper.java [hunk 1]
```diff
@@ -108,7 +107,7 @@ public class PresentationHelper {
      */
     public PresentationHelper(@NonNull Context context,
                               @NonNull PresentationSession presentationSession,
-                              @NonNull DataTransport transport,
+                              @Nullable DataTransport transport, //TODO: make it NonNull
```
Comment (eduardo-flores) [inline identity/src/main/java/com/android/identity/PresentationHelper.java]: Please revert this file, it is not required to have nullable transport.

File: appholder/src/main/java/com/android/mdl/app/document/DocumentManager.kt [hunk 2]
```diff
@@ -1247,7 +1247,7 @@ class DocumentManager private constructor(private val context: Context) {
         )
     }
 
-    fun showData(document: Document) {
+    fun showData(document: Document): CredentialDataResult.Entries? {
```
Comment (eduardo-flores) [inline appholder/src/main/java/com/android/mdl/app/document/DocumentManager.kt]: We can remove this method as we moved it to TransferManager class

File: appholder/src/main/java/com/android/mdl/app/documentdata/DocumentDataReader.kt [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+                MajorType.ARRAY, MajorType.MAP -> return false
+                else -> {
+                }
+            }
+        }
+        return true
+    }
+
+    companion object {
+        private const val MDL_DOCTYPE = "org.iso.18013.5.1.mDL"
```
Comment (eduardo-flores) [inline appholder/src/main/java/com/android/mdl/app/documentdata/DocumentDataReader.kt]: Should we import these constants from DocumentData object?

**Review by eduardo-flores — COMMENTED** [review]
Please take a look at the 3 comments

File: identity/src/main/java/com/android/identity/PresentationHelper.java (hunk 1 above)
Comment (mitrejcevski) [reply identity/src/main/java/com/android/identity/PresentationHelper.java]: Sure!

File: appholder/src/main/java/com/android/mdl/app/document/DocumentManager.kt (hunk 2 above)
Comment (mitrejcevski) [reply appholder/src/main/java/com/android/mdl/app/document/DocumentManager.kt]: Ah correct, we moved it there and this one got left behind. Will remove it

File: appholder/src/main/java/com/android/mdl/app/documentdata/DocumentDataReader.kt (hunk 3 above)
Comment (mitrejcevski) [reply appholder/src/main/java/com/android/mdl/app/documentdata/DocumentDataReader.kt]: Sure!

**Pull request merged**